Extensions/Polygon.cpp

```diff
@@ -0,0 +1,187 @@
+//--------------------------------------------------------------------------------
+//constexpr float PixelAlphaGain = 255.0;
+constexpr float LoAlphaThreshold = 1.0 / 32.0;
+constexpr float HiAlphaThreshold = 1.0 - LoAlphaThreshold;
+
+//--------------------------------------------------------------------------------
+// Calculates closest distance of point to a vector (with origin at 0,0)
+float TFT_eSPI::calcDistance(tFPoint *aPoint, tFPoint *aVector) {
+    float lH = fmaxf(fminf((aPoint->X * aVector->X + aPoint->Y * aVector->Y) / (aVector->X * aVector->X + aVector->Y * aVector->Y), 1.0f), 0.0f);
+    float lDx = aPoint->X - (aVector->X * lH);
+    float lDy = aPoint->Y - (aVector->Y * lH);
+    return sqrtf(lDx * lDx + lDy * lDy);
+}
+
+//--------------------------------------------------------------------------------
+// Calculates closest distance of point to a vector (with origin at 0,0)
+// Result
+// 0: No intersection
+// 1: Intersection with ascending vector 
+// -1: Intersection with descending vector 
+int TFT_eSPI::checkIntersection(tFPoint *aPoint, tFPoint *aVector) {
+    // Detect intersection with vector to -inf
+    bool lInter = false;
+    int lResult = 0;
+    if ((aVector->Y != 0) && (((aPoint->Y >= 0) && (aPoint->Y <= aVector->Y)) || ((aPoint->Y <= 0) && (aPoint->Y >= aVector->Y)))) {
+        if ((aVector->X != 0) && (aPoint->X != 0)) { 
+            float lV = aVector->Y / aVector->X;
+            float lP = aPoint->Y / aPoint->X;
+            if (aPoint->X > 0)
+                lInter = ((aVector->X < 0) || (fabsf(lP) <= fabsf(lV)));
+            else
+                lInter = ((aVector->X < 0) && (fabsf(lP) >= fabsf(lV)));
+        } else 
+            lInter = (aPoint->X >= aVector->X);
+    }
+    if (lInter) {
+        lResult = (aVector->Y >= 0) ? 1 : -1;
+    }
+    // Serial.printf("%f, %f, %f, %f  lV= %f  lP= %f  res= %d\r\n", aPoint.X, aPoint.Y, aVector.X, aVector.Y, lV, lP,  lResult);
+    return lResult;
+}
+
+//----------------------------------------------------
+// in degrees (0..360)
+void TFT_eSPI::rotatePoint(tFPoint *aPoint, float aAngle) {
+    float lAngle = aAngle * PI / 180;
+    float lX = (aPoint->X * cos(lAngle)) - (aPoint->Y * sin(lAngle));
+    float lY = (aPoint->X * sin(lAngle)) + (aPoint->Y * cos(lAngle));
+    aPoint->X = lX;
+    aPoint->Y = lY;
+}
+
+//--------------------------------------------------------------------------------
+// Rotation in degrees (0..360)
+void TFT_eSPI::fillSmoothPolygon(tFPoints aPoints, float aLineWidth, uint32_t aLineColor, 
```

Comment:
Consider passing `aPoints` by const reference to avoid copying the structure.

Reply:
Hi B3rn475,
thanks for your comment. I just changed the code considering your suggestions.

Reply:
See other comment.